gaeul/source/source-application.c

```diff
@@ -196,12 +196,15 @@ gaeguli_nest_dbus_register (GaeguliNest * nest, GDBusConnection * connection)
 static gboolean
 gaeguli_nest_start (GaeguliNest * nest, const gchar * stream_id,
     const gchar * uri, GaeguliVideoCodec codec, guint bitrate,
-    const gchar * passphrase, GaeguliSRTKeyLength pbkeylen)
+    const gchar * passphrase, GaeguliSRTKeyLength pbkeylen,
+    gboolean use_hw_vaapi_decode)
 {
   g_autoptr (GError) error = NULL;
 
   g_return_val_if_fail (nest->target_stream == NULL, FALSE);
 
+  g_object_set (nest->pipeline, "use-hw-vaapi-decode", use_hw_vaapi_decode, NULL);
```

Comment:
I think _GaeguliVideoCodec_ has already a clue which decoder we should use. e.g. if `GAEGULI_VIDEO_CODEC_H264_VAAPI` is specified, there's no reason to use CPU decoding in the pipeline.

Reply:
Used GaeguliVideoCodec details to choose vaapi decoding when GAEGULI_VIDEO_CODEC_H264_VAAPI is mentioned.

Reply:
I believe these changes are redundant now as the hardware decoder is chosen if it available.